core/attachments/S3AttachmentClient.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+import io
+
+from discord import Message
+from minio import Minio
+from minio.commonconfig import Tags
+
+from core.attachments.AttachmentHandler import IAttachmentHandler
+
+
+class S3AttachmentHandler(IAttachmentHandler):
+    def __init__(
+        self,
+        endpoint: str,
+        access_key: str | None = None,
+        secret_key: str | None = None,
+        region: str | None = None,
+        bucket: str | None = "modmail-attachments",
+    ) -> None:
+        """
+        Initialize the S3AttachmentHandler with the given access key, secret key, and endpoint.
+
+        Parameters
+        ----------
+        access_key : str | None
+            The access key for the S3 bucket.
+        secret_key : str | None
+            The secret key for the S3 bucket.
+        endpoint : str
+            The endpoint for the S3 bucket.
+        """
+        self.bucket = bucket
+        self.region = region
+
+        self.client = Minio(
+            endpoint, access_key=access_key, secret_key=secret_key, secure=False, region=region
+        )
+        if not self.client.bucket_exists(self.bucket):
+            self.client.make_bucket(self.bucket)
+
+    async def upload_attachments(self, message: Message) -> list[dict]:
+        """
+        Upload attachments from a given message to the S3 bucket.
+
+        Parameters
+        ----------
+        message : Message
+            The message containing the attachments to upload.
+
+        Returns
+        -------
+        list[dict]
+            A list of dictionaries containing information about the uploaded attachments.
+        """
+        attachments = []
+        tags = Tags.new_object_tags()
```

Comment:
Should include tag for bot ID to help people that are sharing a bucket between multiple bots that they host